external/kafka/src/test/scala/org/apache/spark/streaming/kafka/DirectKafkaStreamSuite.scala

```diff
@@ -313,4 +356,24 @@ class DirectKafkaStreamSuite
 object DirectKafkaStreamSuite {
   val collectedData = new mutable.ArrayBuffer[String]()
   var total = -1L
+
+  class InputInfoCollector extends StreamingListener {
+    val streamIdToNumRecordsSubmitted = new ArrayBuffer[Map[Int, Long]]()
```

Comment:
If you are just going to test the total number of records received, why do you need an arraybuffer?